test/e2e/operator_test.go

```diff
@@ -2723,6 +2723,93 @@ func TestTunableRouterKubeletProbesForCustomIngressController(t *testing.T) {
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+	}
+
+	// Clean up our new Ingress Controller after we are done.
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Wait for new ingress controller to come online.
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// Create services that could collide with the new ingress controller's naming convention for loadbalancer and nodeport.
+	// TRICKY: Our private ingress controller will reconcile (delete) extra loadBalancerServices, even though it isn't one.
+	echoLoadBalancerServiceName := types.NamespacedName{
+		Name:      "router-" + icName.Name,
+		Namespace: "openshift-ingress",
+	}
+	echoLoadBalancerService := buildEchoService(echoLoadBalancerServiceName.Name, echoLoadBalancerServiceName.Namespace, nil)
+	if err := kclient.Create(context.TODO(), echoLoadBalancerService); err != nil {
+		t.Fatalf("failed to create service %s: %v", echoLoadBalancerServiceName, err)
+	}
+
+	echoNodeportServiceName := types.NamespacedName{
+		Name:      "router-nodeport-" + icName.Name,
+		Namespace: "openshift-ingress",
+	}
+	echoNodeportService := buildEchoService(echoNodeportServiceName.Name, echoNodeportServiceName.Namespace, nil)
+	if err := kclient.Create(context.TODO(), echoNodeportService); err != nil {
+		t.Fatalf("failed to create service %s: %v", echoNodeportServiceName, err)
+	}
+
+	// Clean up our new services after we are done.
+	defer func() {
+		if err := kclient.Delete(context.TODO(), echoLoadBalancerService); err != nil {
+			t.Fatalf("failed to delete service %s: %v", echoLoadBalancerServiceName, err)
+		}
+		if err := kclient.Delete(context.TODO(), echoNodeportService); err != nil {
+			t.Fatalf("failed to delete service %s: %v", echoNodeportServiceName, err)
+		}
+	}()
+
+	ic, err := getIngressController(t, kclient, icName, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	// Annotate the ingress operator, to trigger a reconcilation to determine if our service is deleted.
+	// This may not be needed, but it ensures a reconcilation occurs ASAP.
+	if ic.Annotations == nil {
+		ic.Annotations = map[string]string{}
+	}
+	ic.Annotations["ingress.operator.openshift.io/e2e-name-collision-test"] = ""
+
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatal(err)
+	}
+
+	// Wait to see if our service gets deleted by the operator due to name collision.
+	oldLoadBalancerUID := echoLoadBalancerService.UID
+	oldNodePortUID := echoNodeportService.UID
+	wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		// First Check our LoadBalancer Service.
+		if !assertServiceNotDeleted(t, echoLoadBalancerServiceName, oldLoadBalancerUID) {
+			return true, nil
+		}
```

Comment:
`!assertServiceNot` is a double-negative.  We should avoid double-negatives, to improve clarity, if doing so isn't too difficult.  In this case, you don't really need the return value because `assertServiceNotDeleted` calls `f.Fatalf` itself if the assertion fails, so you can remove the double-negative and simplify the logic:
```suggestion
		assertServiceNotDeleted(t, echoLoadBalancerServiceName, oldLoadBalancerUID)
```

Reply:
Agree. Removed the if statement all together so it's just 1 negative now.